cgsmiles/graph_layout.py

```diff
@@ -4,10 +4,46 @@
 from collections import defaultdict, OrderedDict
 import math
 import numpy as np
+from numpy import unravel_index
 import scipy.optimize
 from scipy.spatial.distance import pdist, squareform
 import networkx as nx
 
+def align_principal_axis_with_y(positions):
+    # Step 1: Compute the covariance matrix
+    cov_matrix = np.cov(positions.T)
+
+    # Step 2: Find eigenvalues and eigenvectors of the covariance matrix
+    eigenvalues, eigenvectors = np.linalg.eig(cov_matrix)
+
+    # Step 3: The principal axis is the eigenvector corresponding to the largest eigenvalue
+    principal_axis = eigenvectors[:, np.argmax(eigenvalues)]
+
+    # Step 4: Calculate the rotation angle to align the principal axis with the y-axis
+    # We want to rotate the principal axis to align it with the vector [0, 1] (the y-axis)
+    angle = np.arctan2(principal_axis[0], principal_axis[1])
+
+    # Step 5: Create the rotation matrix to rotate by the negative of this angle
+    rotation_matrix = np.array([[np.cos(-angle), -np.sin(-angle)], 
+                                [np.sin(-angle), np.cos(-angle)]])
+
+    # Step 6: Apply the rotation matrix to all positions
+    rotated_positions = positions @ rotation_matrix.T
+
+    return rotated_positions
```

Comment:
See also https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.transform.Rotation.html#scipy.spatial.transform.Rotation
Bit overkill maybe

Reply:
I saw that as well, but honestly let's keep it simple.

Reply:
I externalized the rotation matrix though